### DISCUSSION TITLE
/guide/docker.html

### DISCUSSION THREAD

**Discussion by giscus** [Mereithhh/vanblog-comment#4] [Announcements]
# /guide/docker.html

Van Blog 的官方网站

https://vanblog.mereith.com/guide/docker.html

**Comment by jingyigong** [comment]
从阮老师那边知道的这个宝贝VanBlog，一款简洁实用的个人博客解决方案，作者的理念我完全赞同，希望能保持初心，创造辉煌。

**Reply by Mereithhh** [reply]
感谢支持！我会努力的！如果有什么意见或建议您可以尽可能提出来，我会及时处理采纳的。

**Comment by msjs2333** [comment]
虽然很丢脸，但是作为一个小白，希望出一个群晖nas的图形化docker的部署教程
>原谅我才上初中

**Reply by Mereithhh** [reply]
初中就开始搞了这些了呀！有前途！没问题，我回头补一个放到项目文档里。

**Comment by ghost** [comment]
在腾讯云轻量服务器上成功部署VanBlog
用nginx-proxy-manager做的反向代理 
https://www.fenghua.fun

**Reply by Mereithhh** [reply]
感谢支持！如果有什么意见或建议您可以尽可能提出来，我会及时处理采纳的。

**Reply by ghost** [reply]
亮色主题下，编辑器预览效果是暗色的，预览效果不明显，能不能更改编辑器预览效果主题，就像这个在线编辑器https://markdown.devtool.tech/app
![](https://tu.zixi.run/i/2022/08/29/iudwt5.webp)
![](https://tu.zixi.run/i/2022/08/29/iwv24d.webp)

**Reply by Mereithhh** [reply]
亮色模式下编辑器预览也是亮色的呀，编辑器的预览模式会随着页面主题的切换而切换。

您可以手动点几下切换主题按钮，可能就正常了。如果还不正常，麻烦您提个 issue ，详细描述一下情况。

![image](https://user-images.githubusercontent.com/22872368/187120931-2afb6812-f600-4c93-81fa-92f61480d5e9.png)

**Reply by Mereithhh** [reply]
刚刚在发版，可能是发版过程中导致的，您可以情况一下停用缓存刷新一下看看。

**Reply by ElijahWmj** [reply]
好哥哥，你的搜索报错了。

**Reply by Mereithhh** [reply]
您指的哪里的搜索呀？

**Comment by redgreat** [comment]
觉得docker或kubernetes需要更多资源还要学那么多，麻烦老板加个虚拟机直接部署的包吧~~

**Reply by Mereithhh** [reply]
感谢建议！

`VanBlog` 内部由很多微服务组成，直接部署到裸机环境可能会由于硬件、系统版本不同、软件不同而出现很多意料之外的问题，容器化可以提供很好的隔离环境，避免因这些差异导致的问题。

实际上容器化是很不会浪费多余的资源的，以 `demo` 站为例，大概只需要 `400M` 内存就可以了。

只是运行容器的话，基本上没什么学习成本，和一键部署也没区别了，迁移和升级都非常方便。

目前还有很多值得完善的地方，所以暂时不会考虑出裸机运行的包或者教程。（做独立包需要考虑很多东西，时间不够呀）

但最近会出一个一键部署脚本（实际上也是自动帮你安装 docker 并启动容器），可以持续关注一下哦

**Reply by redgreat** [reply]
太棒了，docker-compose确实方便，做DB运维的还是不咋接受虚拟机，等自己研究不通再尝试docker，感谢老板，这响应速度太快了~

**Reply by Mereithhh** [reply]
客气了，这是应该的

**Comment by pangxiaobin** [comment]
博主，这个文档是用的开源项目吗来写的么

**Reply by Mereithhh** [reply]
是的，地址： https://vuepress-theme-hope.github.io/

**Comment by zzflsxygod** [comment]
请问用docker-compose部署的时候，如何添加多个域名
文档中说用英文逗号分隔，但是不成功，后台还是会显示请添加域名

试过这样的
> VAN_BLOG_ALLOW_DOMAINS: "www.mydomain.com,192.168.1.2"

也试过这样的，直接无法启动，docker-compose显示未知key
> VAN_BLOG_ALLOW_DOMAINS: "www.mydomain.com","192.168.1.2"

**Reply by Mereithhh** [reply]
没关系了，现在自从 `v0.42.0` 后就不需要这个环境变量了

**Comment by ghost** [comment]
https://vanblog.mereith.com/docker-compose-template.yml它404了！

**Reply by Mereithhh** [reply]
已经修了！谢谢提醒

**Comment by X-brink** [comment]
这个可以离线部署吗，一键部署是在线的吗？

**Reply by Mereithhh** [reply]
是在线的，离线部署也行，参考 docker 部署，手动导入离线 docker 镜像包呗。。

**Comment by Alps-M** [comment]
可以部署到GitHub pages吗

**Reply by Mereithhh** [reply]
从设计上没考虑过这一点，但如果真想做的话，应该有办法做到吧…  可以等后面流水线功能出来后，支持在特定事件发生时执行自定义代码或者发送 webhook，到时候可以自己写一个脚本试试看

**Comment by samtzuss** [comment]
請問有沒有可能將數據庫換成mysql ?

**Reply by Mereithhh** [reply]
暂时不打算支持，以为很多功能还没做完，数据库兼容的优先级并不高。

**Comment by chancat87** [comment]
初一放假在家装了一下试试，一键脚本安装docker等了五分钟以上无反应之后ssh中断了。。昨天才撸的v.ps家的新机器安装按理说dd了系统没问题呀。得空再换个机器试试🤣

**Reply by chancat87** [reply]
刚刚又换了个ion的vps..这次安装到之顺利的。大概是这台服务器我之前部署过docker的原因，看到执行命令读取一下信息就直接开始拉取了。等下我再测试之前买的不成功的机器试试。？不过配置低，512m内存的小鸡，跑docker的话被占满迟早的事，io性能也是个问题，到不如直接nginx跑个wordpress..

**Reply by Mereithhh** [reply]
目前确实比较花性能，性能优化已经在路上了。。。

**Comment by sksksksy** [comment]
需要添加docker仓库吗？使用docker下载mereith/van-blog:latest镜像，提示找不到。

**Reply by Mereithhh** [reply]
不需要，有的啊。https://hub.docker.com/r/mereith/van-blog
